scripts/o3de/o3de/register.py

```diff
@@ -365,19 +375,23 @@ def register_external_subdirectory(json_data: dict,
                                    external_subdir_path: pathlib.Path,
                                    remove: bool = False,
                                    engine_path: pathlib.Path = None,
-                                   project_path: pathlib.Path = None) -> int:
+                                   project_path: pathlib.Path = None,
+                                   gem_path: pathlib.Path = None) -> int:
     """
     :return An integer return code indicating whether registration or removal of the external subdirectory
     completed successfully
     """
-    # If a project path or engine path has not been supplied auto detect which manifest to register the input path with
-    if not project_path and not engine_path:
+    # If a gem path, project path or engine path has not been supplied auto detect which manifest to register the input path with
+    if not gem_path:
+        gem_path = utils.find_ancestor_dir_containing_file(pathlib.PurePath('gem.json'), external_subdir_path)
+    if not project_path:
         project_path = utils.find_ancestor_dir_containing_file(pathlib.PurePath('project.json'), external_subdir_path)
-        if not project_path:
-            engine_path = utils.find_ancestor_dir_containing_file(pathlib.PurePath('engine.json'), external_subdir_path)
+    if not engine_path:
```

Comment:
These should be `elif` blocks in this case, we want it to be mutally exclusive, so that only one "O3DE Object" file of gem.json, project.json or engine.json is updated.

The priority of updates is the gem path is used if it exist, followed, by the project path if it exist, followed by the engine path if it exist.
Finally if none of the paths exist, the registration will happen with the users `~/.o3de/o3de_manifest.json` file